sw/host/opentitanlib/src/test_utils/otp_ctrl.rs

```diff
@@ -66,6 +66,18 @@ impl OtpParam {
     pub fn write_param(jtag: &dyn Jtag, param: DaiParam, data: &[u32]) -> OtpDaiResult<()> {
         let OtpParamMmap { byte_addr, size } = param.mmap();
 
+        // Check if the word has been written already.
+        let mut curr_value: Vec<u32> = Vec::new();
+        for _ in 0..(size / 4) {
+            curr_value.push(0u32);
+        }
+        Self::read_param(jtag, param, curr_value.as_mut_slice())?;
+        for word in curr_value.iter() {
+            if *word != 0u32 {
+                return Err(OtpDaiError::WriteErrorAlreadyWritten { value: *word });
```

Comment:
The error message here could be confusing for 64-bit parameters if the set value was, e.g. `0xFFFFFFFF00000000` and the message is

`writing to OTP failed since field has already been written (with value: 4294967295)`
(or)
`writing to OTP failed since field has already been written (with value: FFFFFFFF)`

Reply:
Since `read_param` already always returns 32-bit words, regardless of access granule, I kept the error message to display 32-bit words. But good point, we could revisit this in future if needed.